test/functional/p2p_segwit.py

```diff
@@ -869,7 +875,7 @@ def test_block_malleability(self):
     @subtest  # type: ignore
     def test_witness_block_size(self):
         # TODO: Test that non-witness carrying blocks can't exceed 1MB
-        # Skipping this test for now; this is covered in p2p-fullblocktest.py
+        # Skipping this test for now; this is covered in feature_block.py
```

Comment:
(Per `git show ca6523d0c8` (thanks for the commit reference!) this was changed in January 2018, is it still accurate, covered in `feature_block.py`?)

Reply:
Yes, I think the relevant part is here:

https://github.com/bitcoin/bitcoin/blob/fdd80b0a53b4af0b29cb6e03118e2456d053a757/test/functional/feature_block.py#L315-L337

These blocks (as all others in feature_block.py, IIRC) don't have a witness commitment, so they are "non-witness carrying blocks". For those the serialized size equals to the block weight divided by WITNESS_SCALE_FACTOR (4), i.e. with MAX_BLOCK_WEIGHT (4000000) this results in the 1MB size limit being checked. Note that these test parts were changed recently in #22378 to switch to weight-based accounting, the constant used was formerly named `MAX_BLOCK_BASE_SIZE`.